### DISCUSSION TITLE
WinPaletter and registry configuration imports for easy color set up/changing (download)

### DISCUSSION THREAD

**Discussion by DAPOPOBEFASTONYOAZZ** [ALTaleX531/OpenGlass#72] [General]
### DISCLAIMER
**Use the registry import keys at your own risk! I am not responsible for damage caused to your system. You acknowledge this risk that these registry keys may be incompatible with your system when downloading these files.**

### Introduction
It seems like a quite a few people have a hard time setting up their configuration for OpenGlass. While I was reluctant to release my files at first due to incompatibility risks, I figured it would be good and easy help for the community to share to get a default set up configuration.

I'm also someone who likes to change their window border colors seasonally or monthly based on themes I like to pick, but it can be quite the pain in the ass to make these changes in the registry. Instead, I decided to make registry files to make changing colors a breeze! I wanted to share these files for anybody who wanted pre-set files to make setting up OpenGlass easy either for the first time, or anybody who would like to change their colors in 5 seconds rather than 5 minutes.

I have included 48 unique colors, which are the 24 default color choices offered in Windows Vista and Windows 7, also including the 24 colors in their transparency disabled appearance as well, if you prefer that style. If you would like to make your own custom color, **these files only support the provided colors.** You must know how to configure the registry for your own custom color and apply your own settings to StartIsBack.

### Important
You will need WinPaletter to assist you in this process, which you can download on GitHub. These themes are compatible with WinPaletter version 1.0.9.3. Old versions are not compatible with these theme files and new versions may also break compatibility with existing theme files.

These themes are also built with compatibility for StartIsBack only, it does NOT support OpenShell!

These registry imports will complete the color changing process of the provided colors and will automatically change these settings to match your theme once once applied:

- Inactive window transparency (does not effect any of the transparency disabled themes, this setting is already set in WinPaletter if you use a transparency disabled theme.)
- Color intensity/opacity
- Legacy colors of Windows (such as text selection) to maintain authenticity with the Windows Vista and 7 Aero themes.
- StartIsBack color and intensity/opacity settings

Lastly, the WinPaletter themes use Windows 8's default purple color for Windows 10 elements (action panel, notification pop-ups and more,) it only changes the color of the title bar. If you would like your Windows 10 elements to be colored based on the colors of the theme, WinPaletter allows you to easily automatically generate accent colors based on the theme you choose. You must go into window colors click to generate these colors before applying the theme if you wish to change these colors. I have highlighted this option in the picture below so you can locate it easily.
![4](https://github.com/user-attachments/assets/6dd080d7-6d8f-4fef-bfd2-2da2a7bf3679)

### Step 1 - only if you do not yet have OpenGlass configured or would like to change to this default configuration I have provided. If you do, then you can skip to step 2.
Install OpenGlass. After installation, download "Default Config.zip" and extract the file. After you extract the file, double click the "Default Config.reg" file to merge with your registry. This will establish default settings consistent to those with standard Windows Aero. For best effects, you will need to download and use a Windows Aero theme. I recommend Aero10.

### Step 2
Download WinPaletter (hosted from another GitHub repository, not offered in this download,) the attached WinPaletter theme files AND the registry color keys. After this step is completed, you can ignore both step 1 and 2 for future changes.

### Step 3
Extract the files and pick WinPaletter theme you would like to use first. **Important: this must be done before selecting a registry key entry to merge.** This is what you will see after applying a theme you choose. Don't panic if things don't look right yet, there's more you need to do.
![1](https://github.com/user-attachments/assets/8d3d6a46-33a8-46a7-9cca-3287154f683d)

### Step 4
After choosing and applying a WinPaletter theme, select the matching registry key entry and double click on it to merge it with your registry. You will be prompted by User Account Control (UAC) to accept or deny the changes. Click on accept. You will then be greeted by another prompt asking you if you are sure that you would like to modify the registry. Click on yes. **Please remind yourself of the disclaimer at the beginning of this post before performing this action.**

After merging the registry files, you will see this. Again, things won't look right just yet. There's one more step.
![2](https://github.com/user-attachments/assets/9f174290-1ce7-4d74-b207-159349dc7005)

### Step 5
These changes are not shown at first after merging the registry entries. To get your new colors to update after merging your registry keys, the easiest way to do this is triggering another UAC prompt, like clicking on the registry key entry you already applied as if you were starting Step 3 again. Except this time, you do not need to merge, so instead at the UAC prompt, click on deny. Your desktop will be refreshed and your changes will be shown. Alternatively, you can log out of your account and log back in.

After this, you should now see all the elements of the theme applied, like so:
![3](https://github.com/user-attachments/assets/f4650b32-f916-435c-b23f-479ae401ba9f)

I hope this helps make it easier and more accessible for newbies.

**Downloads**
OpenGlass configuration: [Default Config.zip](https://github.com/user-attachments/files/18129304/Default.Config.zip)
WinPaletter Themes: [Color Schemes.zip](https://github.com/user-attachments/files/18129320/Color.Schemes.zip)
Registry color configurations: [Registry Colors.zip](https://github.com/user-attachments/files/18129331/Registry.Colors.zip)

**Comment by heathercat123** [comment]
This nuked by C drive icon for whatever reason. I also get issues with the colors themselves: Windows 7's Pumpkin color is too dark, switching from Vista orange (remember that 7's didn't work) to 7 Twilight kept the inactive titlebar color and switching to any color at this point didn't affect the inactive color.

**Reply by DAPOPOBEFASTONYOAZZ** [reply]
> This nuked by C drive icon for whatever reason. I also get issues with the colors themselves: Windows 7's Pumpkin color is too dark, switching from Vista orange (remember that 7's didn't work) to 7 Twilight kept the inactive titlebar color and switching to any color at this point didn't affect the inactive color.

Oops - I realize that I may have left some icons left over in the themes that I meant to take out. You can change this in WinPaletter to restore the default icon.

Yes, I realize some colors after making this post weren't accurate (Taupe was one of them,) and this was also made with an older version of OpenGlass as well, so this is outdated. It relies on some registry keys that weren't yet extended, so it may not be accurate anymore. I'll need to later put together these color schemes again for Legacy 2.0.

**Reply by DAPOPOBEFASTONYOAZZ** [reply]
> This nuked by C drive icon for whatever reason. I also get issues with the colors themselves: Windows 7's Pumpkin color is too dark, switching from Vista orange (remember that 7's didn't work) to 7 Twilight kept the inactive titlebar color and switching to any color at this point didn't affect the inactive color.

I meant to mention too, that if you're using Windows 7 colors with Legacy 2.0, they will always be darker in these current versions. This is because `ColorizationBlurBalanceOverride` actually used to be what was set by `GlassOpacity`. I had the same issue before I discovered that `ColorizationBlurBalanceOverride` had been added.

These will eventually be fixed. I've changed them on my local machine, just hadn't got around to uploading them.

**Reply by heathercat123** [reply]
> Oops - I realize that I may have left some icons left over in the themes that I meant to take out. You can change this in WinPaletter to restore the default icon.
Thanks for telling me! This also fixed a few control panel icons,

Also, deleting OpenGlass' registry keys then applying the WinPaletter themes fixed the inactive color.
